app/src/main/java/org/mozilla/focus/home/HomeFragment.kt

```diff
@@ -228,9 +235,13 @@ private class HomeTileAdapter(
 
     override fun getItemCount() = tiles.size
 
-    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int) = TileViewHolder(
-            LayoutInflater.from(parent.context).inflate(R.layout.home_tile, parent, false)
-    )
+    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): TileViewHolder {
+        var layout = R.layout.bundled_home_tile
+        if (viewType == CUSTOM_TILE_VIEW_TYPE) {
```

Comment:
nit: similar idea to the above: you want to explicitly specify all the states you expect this to be in (and crash if you're not in an expected state!).

Also, you generally want to avoid mutability where possible because someone later can change a value you depend on.

In this case, you can:
```kotlin
val layout = when (viewType) {
```